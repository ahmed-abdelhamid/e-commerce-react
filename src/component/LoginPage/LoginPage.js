import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withFormik, Form, Field } from 'formik';
import * as Yup from 'yup';
import { history } from '../Routers/AppRouter';
import { login } from '../../action/auth';
import authUsers from '../../authUser';

export class LoginPage extends Component {
  state = {
    usernamePlaceholder: 'Username',
    passowrdPlaceholder: 'Password',
  };

  handleFocus = (e) => {
    (e.target.name === 'username') ? (
      this.setState(() => ({ usernamePlaceholder: '' }))
    ) : (
      this.setState(() => ({ passowrdPlaceholder: '' }))
    );
  };

  handleBlur = (e) => {
    (e.target.name === 'username') ? (
      this.setState(() => ({ usernamePlaceholder: 'Username' }))
    ) : (
      this.setState(() => ({ passowrdPlaceholder: 'Password' }))
    );
  };

  render() {
    const { touched, errors, values } = this.props;
    return (
      <div className="container text-center">
        <Form className="login-form">
          <h1 className="login-form__title">Login Admin</h1>
          <div className="form-group">
            { touched.username && errors.username && <div className="error">{errors.username}</div> }
            <Field
              type="text"
              className="form-control"
              name="username"
              value={values.username}
              placeholder={this.state.usernamePlaceholder}
              onFocus={this.handleFocus}
              onBlur={this.handleBlur}
            />
          </div>
          <div className="form-group">
            { touched.password && errors.password && <div className="error">{errors.password}</div> }
            <Field
              type="password"
              className="form-control"
              name="password"
              value={values.password}
              placeholder={this.state.passowrdPlaceholder}
              onFocus={this.handleFocus}
              onBlur={this.handleBlur}
            />
          </div>
          <button type="submit" className="btn btn-primary">
            Login
          </button>
        </Form>
      </div>
    );
  }
};

const FormikLoginPage = withFormik({
  mapPropsToValues() {
    return {
      username: '',
      password: ''
    };
  },
  validationSchema: Yup.object().shape({
    username: Yup.string().required(),
    password: Yup.string().required()
  }),
  handleSubmit(values, { props, setErrors }) {
    authUsers.map((user) => {
      if (user.username === values.username && user.password === values.password) {
        props.login(user);
        history.push('/dashboard');
      } else {
        setErrors({
          username: 'Wrong username or password'
        })
      }
    });  
  }
})(LoginPage);

const mapDispatchToProps = (dispatch) => ({
  login: (uid) => dispatch(login(uid))
});

export default connect(undefined, mapDispatchToProps)(FormikLoginPage);