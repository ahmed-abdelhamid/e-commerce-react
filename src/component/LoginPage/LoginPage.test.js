import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { FormikLoginPage } from './LoginPage';

test('should render LoginPage page', () => {
  const wrapper = shallow(<FormikLoginPage />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});