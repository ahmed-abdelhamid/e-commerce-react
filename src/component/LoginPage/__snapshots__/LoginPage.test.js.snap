// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage page 1`] = `
<Formik
  enableReinitialize={false}
  handleSubmit={[Function]}
  initialValues={
    Object {
      "password": "",
      "username": "",
    }
  }
  isInitialValid={false}
  onSubmit={[Function]}
  render={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
  validationSchema={[Function]}
/>
`;
